### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by moyogo** [typesupply/opentype-feature-intro#16]
What’s the license?

**Comment by moyogo** [comment]
Oops, sorry. I just saw http://opentypecookbook.com/acknowledgements.html.
It would still be useful to see this first in the repository.

**Comment by ldo** [comment]
Pity it’s a [non-Free](http://ldo17.tumblr.com/post/18544016356/creative-commons-dos-donts) licence.

**Comment by typesupply** [comment]
> Pity it’s a non-Free licence.

Yeah. What a pity. How dare I spend over a month writing this and then not give it away for free. Oh wait, I did.

**Comment by ldo** [comment]
“No charge” is not the same as [Free](http://ldo17.tumblr.com/post/18544016356/creative-commons-dos-donts).

**Comment by typesupply** [comment]
What exactly is the license preventing you from doing?

**Comment by ldo** [comment]
Problem areas are [things like](http://blog.okfn.org/2013/01/08/consequences-risks-and-side-effects-of-the-license-module-non-commercial-use-only-2/):
* Schools, universities, teaching institutions
* Public institutions
* A blog that gets advertising revenues
* Big corporations with more lawyers than you.

**Comment by Jolg42** [comment]
@ldo I can see what problems you can have in some cases, but if you have a specific need it's better to ask the author directly if you can have the right to do it.
In general it's not a problem, people are nice and give a lot of stuff for free, but here your comments are not helping, it seems like you don't see the amount of time @typesupply invested in it…!

Please remember that behind the @ there are humans 😉

@typesupply By the way, thank you for your work! 👍

**Comment by ldo** [comment]
Look at it this way: NC is a field-of-use restriction (see section 6 [here](https://opensource.org/osd-annotated)). It is not the best way to contribute to the Free Culture community.

It’s understandable that you don’t want big companies who want something for nothing to come in rip you off. That’s why we have [CC-BY-SA](http://creativecommons.org/licenses/by-sa/4.0/).

**Comment by typesupply** [comment]
> Big corporations with more lawyers than you.

The thing is, I have interacted with lawyers from big corporations _and_ Free software pseudo-lawyers. Without a doubt, the real lawyers have been far easier to work with and far more understanding of artist/designer/author/developer rights. Yes, they do want to spend as little money as possible, but in my dozens of interactions with them they have never tried to convince me that I should give them my work for 100% free. The pseudo-lawyers, on the other hand, have never recognized the value of the work that they are attempting to fre—er—Free™ and have been really, incredibly annoying. Not once has one of these advocates said, "Ah, okay, I see your point." It's always, "But...you are wrong...Free!" repeated to infinity.

Here is an actual number for this particular project: If you take the amount of money that has been donated to my OpenType Cookbook and subtract if from the domain registration + paying for the development of the HTML/CSS + other miscellaneous expenses, I'm at around **-$900**. That's $900 of my own money. That doesn't include the value of time I spent writing the book, which was about 1/12th of my work year, or the value of time of the experts who graciously reviewed the book. It also doesn't include the value of my decade+ of hard won knowledge gained through trial and error that gave me the knowledge that is in the book. I'm completely cool with all of this because this is information that I think is beneficial to the craft of type design. I gladly paid, and would pay again, for the wonderful work that cost me actual money on this project. Those efforts added value to the project and deserved compensation. I knew it would be a net loss before I started. You can look at my list of repositories and see that I have made this same decision a lot in the past.

Over the last month I've had to make decisions about what I should do with three major codebases and one book-like thing. I decided that the benefits of open sourcing the code don't outweigh the annoyance of the "Make it Free!" comments plus the frustrations of developers who make use of my work but don't contribute anything of their own. So, that code is going to remain closed. 10 years ago I would have open sourced it without a second thought. Now, not so much. The book-like thing is something that I'm still thinking through. It's a collection of well thought out ideas that I don't have time to build. I was thinking that I'd put them in the public domain and let someone else take them from there. But again, is it worth the aggravation? At this point, I can't say that it is.

These nit-picking lectures about Free™-compatibility are completely counter productive when you look at the bigger picture. I mean, did you honestly think your pithy "Pity" comment was going to have a positive effect?

tl:dr; Things don't come from nowhere. Goodness gracious I'm tired of these comments.

**Comment by ldo** [comment]
This is getting rather severely off-topic, so just to make it clear: you may have a certain meaning of “non-commercial” in mind, but if it ever came to a legal decision, a court [may not agree with you](https://www.techdirt.com/articles/20140326/11405526695/german-court-says-creative-commons-non-commercial-licenses-must-be-purely-personal-use.shtml).

**Comment by typesupply** [comment]
> This is getting rather severely off-topic,

No. It's entirely on topic. You showed up and pettily complained about something trivial. You made no constructive comment about the work itself. It was just a run-of-the-mill "The license isn't pure enough!" hit and run comment. These are the very things that are making me not want to open source or public domain my work anymore. These kinds of snipes are completely counter productive when you consider the larger goal of making stuff more open and available. Largely as a result of these kinds of comments, instead of releasing three new projects under a "pity it's a non-Free license." I'm simply not going to release them publicly so that I don't have to waste my time responding to comments like this on those projects.

I mean, it's fine to be a zealot for Free™ or Libre or whatever you want to call it. I really have no problem with people choosing that license model for their own work. But if someone has _clearly made a decision about their license_ it's probably best not to try to annoy them into submission even if you consider their choice to be a pity.

> so just to make it clear: you may have a certain meaning of “non-commercial” in mind, but if it ever came to a legal decision, a court may not agree with you.

...

>> "But...you are wrong...Free!" repeated to infinity.

**Comment by behdad** [comment]
Tal, please don't let these kind of issues affect your decisions-making. For anyone who doesn't like your work (for any reason), there are 200 who really appreciate you releasing all the stuff you do.  Just ignore and move on if you don't want to discuss.  You don't even need to justify your choice of license.

**Comment by ldo** [comment]
[Yet another example](http://arstechnica.com/tech-policy/2016/09/creative-commons-licenses-under-scrutiny-what-does-noncommercial-mean/) of the pitfalls...

**Comment by moyogo** [comment]
@ldo Your frustration with the NC limbo is understandable but the way you’re communicating is just creating more frustration. I think it’s better to let it go.

**Comment by ldo** [comment]
I don’t know what your idea of a project where people are supposed to collaborate looks like, but this certainly isn’t it. I tried to point out an issue dispassionately, concentrating on the facts rather than egos or personalities, and in return I have been the target of personal attack after personal attack.

Well, I have had enough of all of you. Goodbye.

**Comment by typesupply** [comment]
> Well, I have had enough of all of you. Goodbye.

Goodbye!
